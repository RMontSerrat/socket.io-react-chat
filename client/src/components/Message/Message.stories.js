import React from 'react';
import { storiesOf } from '@storybook/react';
import Message from './Message';

storiesOf('Message', module)
  .add('default', () => (
    <div style={{ padding: '20px' }}>
      <Message userName="Romulo" date={new Date()}>Fala galera</Message>
    </div>
  ))
  .add('without userName', () => (
    <div style={{ padding: '20px' }}>
      <Message userName="Servidor" date={new Date()}>Romulo entrou na sala...</Message>
    </div>
  ));

