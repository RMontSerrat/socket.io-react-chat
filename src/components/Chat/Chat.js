import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import MessageGroup from '../MessageGroup';
import MessageBar from '../MessageBar';

const MessageBarContainer = styled.div`
  padding-top: 15px;
`
const ChatContainer = styled.div`
  display: flex;
  flex-direction: column;
`
const StyledMessageGroup = styled(MessageGroup)`
  overflow-x: auto;
  height: 100%;
`

const Chat = ({ onSubmit, children }) =>
  <ChatContainer>
    <StyledMessageGroup>
      {children}
    </StyledMessageGroup>
    <MessageBarContainer>
      <MessageBar onSubmit={onSubmit} />
    </MessageBarContainer>
  </ChatContainer>

Chat.propTypes = {
  onSubmit: PropTypes.func,
  children: PropTypes.node,
}

export default Chat;