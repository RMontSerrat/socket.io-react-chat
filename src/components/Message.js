import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  color: ${props => props.theme.colorPrimary};
  font-family: ${props => props.theme.fontFamily};
  padding: 5px;
  background-color: ${props => props.active ? props.theme.colorLightGreen : 'transparent'};
`

const DateContainer = styled.span`
  display: block;
  padding-right: 10px;
`
const Name = styled.strong`
  padding-right: 5px;
`

const Message = ({ children, userName = 'Servidor', date, active }) =>
  <Container active={active}>
    <DateContainer>[{new Date(date).toLocaleTimeString()}]</DateContainer>
    <Name>{userName}:</Name>
    {children}
  </Container>

export default Message;