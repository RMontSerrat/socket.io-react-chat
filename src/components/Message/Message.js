import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  color: ${props => props.theme.colorPrimary};
  font-family: ${props => props.theme.fontFamily};
  padding: 5px;
`

const DateContainer = styled.span`
  display: block;
  padding-right: 10px;
`
const Name = styled.strong`
  padding-right: 5px;
`

const Message = ({ children, userName, system, date }) =>
  <Container>
    <DateContainer>[{new Date(date).toLocaleTimeString()}]</DateContainer>
    <Name>{system ? 'Servidor' : userName}:</Name>
    {children}
  </Container>

export default Message;