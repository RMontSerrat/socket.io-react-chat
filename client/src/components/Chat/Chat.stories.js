import React from 'react';
import { storiesOf } from '@storybook/react';
import Chat from './Chat';
import Message from '../Message';

const messages = [{
    msg: 'Romulo entrou na sala',
    date: new Date(),
  }, {
    userName: 'Romulo',
    msg: 'Fala galera',
    date: new Date(),
}, {
  msg: 'Zé entrou na sala',
  date: new Date(),
  system: true,
}, {
  userName: 'Zé',
  msg: 'e aiiii....',
  date: new Date(),
}]

const onSubmit = value => { 
  console.log(value);
}

storiesOf('Chat', module)
  .add('default', () => (
    <div style={{ backgroundColor: '#025a3f', padding: '20px' }}>
      <Chat onSubmit={onSubmit}>
        {messages.map(({ msg, ...item }) => <Message {...item}>{msg}</Message>)}
      </Chat>
    </div>
  ));
