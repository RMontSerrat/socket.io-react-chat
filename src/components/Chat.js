import React from 'react';
import styled from 'styled-components';
import MessageGroup from './MessageGroup';
import Message from './Message';
import MessageBar from './MessageBar';

const Title = styled.h1`
  font-family: ${props => props.theme.fontFamily};
  font-size: 30px;
  color: ${props => props.theme.colorWhite};
  grid-area: title;
`
const MessageBarContainer = styled.div`
  padding-top: 15px;
  grid-area: input-container;
`
const ChatContainer = styled.div`
  display: grid;
  grid-template-areas: 'title' 'message-group' 'input-container';
  grid-template-rows: auto 60% auto;
`
const StyledMessageGroup = styled(MessageGroup)`
  overflow-x: auto;
  grid-area: message-group;
`

const Chat = ({ onSubmit, data, name }) =>
  <ChatContainer>
    <Title>Chat {name}</Title>
    <StyledMessageGroup>
      {data.map(({ children, ...item }) => <Message {...item}>{children}</Message> )}
    </StyledMessageGroup>
    <MessageBarContainer>
      <MessageBar onSubmit={onSubmit} />
    </MessageBarContainer>
  </ChatContainer>


export default Chat;