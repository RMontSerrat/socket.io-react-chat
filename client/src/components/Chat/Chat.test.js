import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import Chat from "./Chat";
import Message from '../Message';
import 'jest-styled-components'

afterEach(cleanup);
describe('<Chat />', () => {
  it("should render", () => {
    const { asFragment } = render(
      <Chat>
        <Message date={new Date('2019-01-01')} userName="Romulo">boa noite.</Message>
      </Chat>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render with onSubmit", () => {
    const onSubmit = jest.fn();
    const { getByTestId } = render(
      <Chat onSubmit={onSubmit}>
        <Message date={new Date('2019-01-01')} userName="Romulo">boa noite.</Message>
      </Chat>
    );
    const form = getByTestId('messageBarSubmit');
    fireEvent.submit(form);
    expect(onSubmit).toHaveBeenCalled();
  });

});