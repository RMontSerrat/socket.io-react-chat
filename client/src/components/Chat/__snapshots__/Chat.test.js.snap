// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chat /> should render 1`] = `
<DocumentFragment>
  .c6 {
  padding: 15px 10px;
  width: calc(100% - 20px);
  border: 0;
  outline: none;
}

.c8 {
  width: 100%;
  font-weight: bold;
  -webkit-transition: all 0.2s ease-in 0s;
  transition: all 0.2s ease-in 0s;
  border: 0;
  cursor: pointer;
  outline: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  width: 250px;
  padding-left: 10px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  overflow-x: auto;
  height: 100%;
}

.c1 > div {
  margin: 10px 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
}

.c3 {
  display: block;
  padding-right: 10px;
}

.c4 {
  padding-right: 5px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          [22:00:00]
        </span>
        <strong
          class="c4"
        >
          Romulo:
        </strong>
        boa noite.
      </div>
    </div>
    <div
      class=""
    >
      <form
        class="c5"
        data-testid="messageBarSubmit"
      >
        <input
          class="c6"
          data-testid="messageBarInput"
          placeholder="Digite sua mensagem aqui..."
          value=""
        />
        <div
          class="c7"
        >
          <button
            class="c8"
            type="submit"
          >
            Enviar
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
